client/src/views/files/FileDetailsModal.vue

```diff
@@ -6,111 +6,251 @@
-          </ion-text>
-        </ion-item>
-        <!-- Last updated -->
-        <ion-item class="body-lg">
-          <ion-label class="file-info-key">
-            {{ $t('FileDetails.stats.updated') }}
-          </ion-label>
-          <ion-text class="file-info-value">
-            {{ $d(entry.updated.toJSDate(), 'long') }}
-          </ion-text>
-        </ion-item>
-        <ion-item class="body-lg">
-          <ion-label class="file-info-key">
-            {{ $t('FileDetails.stats.version') }}
-          </ion-label>
-          <ion-text class="file-info-value">
-            {{ entry.baseVersion }}
-          </ion-text>
-        </ion-item>
-        <!-- Is synchronized -->
-        <ion-item class="body-lg">
-          <ion-label class="file-info-key">
-            {{ $t('FileDetails.stats.isSynced') }}
-          </ion-label>
-          <ion-text class="file-info-value">
-            {{ entry.needSync ? $t('FileDetails.stats.no') : $t('FileDetails.stats.yes') }}
-          </ion-text>
-        </ion-item>
-        <!-- Id -->
-        <ion-item class="body-lg">
-          <ion-label class="file-info-key">
-            {{ $t('FileDetails.stats.id') }}
-          </ion-label>
-          <ion-text class="file-info-value">
-            {{ entry.id }}
-          </ion-text>
-        </ion-item>
+          <div class="file-info-path-value">
+            <ion-text class="file-info-path-value__text body">
+              {{ shortenPath(path) }}
+            </ion-text>
+            <ion-button
+              fill="clear"
+              size="small"
+              id="copy-link-btn"
+              @click="copyPath"
+              v-if="!pathCopiedToClipboard"
+            >
+              <ion-icon
+                class="icon-copy"
+                :icon="copy"
+              />
+            </ion-button>
+            <ion-text
+              v-if="pathCopiedToClipboard"
+              class="file-info-path-value__copied body copied"
+            >
+              {{ $t('FileDetails.stats.linkCopied') }}
+            </ion-text>
+          </div>
+        </div>
       </div>
     </ms-modal>
   </ion-page>
 </template>
 
 <script setup lang="ts">
-import { formatFileSize } from '@/common/file';
+import { writeTextToClipboard } from '@/common/clipboard';
+import { formatFileSize, getFileIcon } from '@/common/file';
 import { MsModal } from '@/components/core';
+import { Folder, MsImage } from '@/components/core/ms-image';
 import { EntryStat, EntryStatFile } from '@/parsec';
-import { IonItem, IonLabel, IonPage, IonText } from '@ionic/vue';
+import { IonButton, IonIcon, IonLabel, IonPage, IonText } from '@ionic/vue';
+import { cloudDone, cloudOffline, copy } from 'ionicons/icons';
+import { defineProps, ref } from 'vue';
 
-defineProps<{
+const props = defineProps<{
   entry: EntryStat;
   path: string;
 }>();
+
+const pathCopiedToClipboard = ref(false);
+
+function shortenPath(fullPath: string): string {
```

Comment:
Just call the function in `common/file.ts`

Reply:
Updated.